storage/blob_test.go

```diff
@@ -273,13 +273,75 @@ func (s *StorageBlobSuite) TestDeleteBlobIfExists(c *chk.C) {
 	blob := randString(20)
 
 	cli := getBlobClient(c)
-	c.Assert(cli.DeleteBlob(cnt, blob), chk.NotNil)
+	c.Assert(cli.DeleteBlob(cnt, blob, nil), chk.NotNil)
 
-	ok, err := cli.DeleteBlobIfExists(cnt, blob)
+	ok, err := cli.DeleteBlobIfExists(cnt, blob, nil)
 	c.Assert(err, chk.IsNil)
 	c.Assert(ok, chk.Equals, false)
 }
 
+func (s *StorageBlobSuite) TestDeleteBlobWithConditions(c *chk.C) {
+	cnt := randContainer()
+	blob := randString(20)
+
+	cli := getBlobClient(c)
+
+	c.Assert(cli.CreateContainer(cnt, ContainerAccessTypePrivate), chk.IsNil)
+	defer cli.deleteContainer(cnt)
+
+	c.Assert(cli.CreateBlockBlob(cnt, blob), chk.IsNil)
+	oldProps, err := cli.GetBlobProperties(cnt, blob)
+	c.Assert(err, chk.IsNil)
+
+	// Update metadata, so Etag changes
+	c.Assert(cli.SetBlobMetadata(cnt, blob, map[string]string{}), chk.IsNil)
+	newProps, err := cli.GetBlobProperties(cnt, blob)
+	c.Assert(err, chk.IsNil)
+
+	// "Delete if matches old Etag" should fail without deleting.
+	err = cli.DeleteBlob(cnt, blob, map[string]string{
+		"If-Match": oldProps.Etag,
+	})
+	c.Assert(err, chk.FitsTypeOf, AzureStorageServiceError{})
+	c.Assert(err.(AzureStorageServiceError).StatusCode, chk.Equals, http.StatusPreconditionFailed)
+	_, err = cli.GetBlob(cnt, blob)
+	c.Assert(err, chk.IsNil)
+
+	// "Delete if matches new Etag" should succeed.
+	err = cli.DeleteBlob(cnt, blob, map[string]string{
+		"If-Match": newProps.Etag,
+	})
+	c.Assert(err, chk.IsNil)
+	_, err = cli.GetBlob(cnt, blob)
+	c.Assert(err, chk.Not(chk.IsNil))
+
+	// Create a non-empty blob and set metadata in the hopes of
+	// triggering a "modified" event that will make an
+	// If-Unmodified-Since request give us a 412.
+	c.Assert(cli.CreateBlockBlobFromReader(cnt, blob, 1, bytes.NewReader([]byte{'-'}), nil), chk.IsNil)
+	c.Assert(cli.SetBlobMetadata(cnt, blob, map[string]string{"foo": "bar"}), chk.IsNil)
+
+	if false {
```

Comment:
I don't really know what to do with this. It would be a useful test, but (afaict) it fails because of a bug in Azure itself. So... just delete this section and pretend we never noticed? Or do the check, but if Azure incorrectly deletes the block, just re-write the block and log something but don't fail the test?